include/nccl_ofi_rdma.h

```diff
@@ -840,6 +840,9 @@ typedef struct nccl_net_ofi_rdma_device {
 #endif
 } nccl_net_ofi_rdma_device_t;
 
+#ifdef __cplusplus
```

Comment:
We never added nvtx domain members into sendrecv structs, so the sendrecv header doesn't pull the tracepoint header in (only the sendrecv TU does), which means we never end up accidentally pulling in the problematic lttng header and its templates while under an `extern "C"` block, so it's not required. 

I don't like this commit. As the commit message mentions, there's two ways to solve this, and I implemented both. But we can just as easily get away with avoiding lttng headers here by including nvtx non-transitively and it should end up cleaner.

Reply:
Alternatively, if those definitions above the `extern "C"` block are truly only used by rdma.c, then they should be moved into the file to enforce that the definitions are private. But so long as it's a `.h` header and not an `.hpp` header, it's really gross to not have every single definition under the same abi.

Reply:
The problem with moving almost everything from rdma.h into the .c file is effectively that the .c file then becomes unmanageable large (it already is, as Eric has pointed out).  And splitting it means bringing those struct declarations back up to a header anyway (perhaps still splitting the header).  We should do those things, but I'd like to do that not today.

In the mean time, since it's harmless, can we make the extern C linkage behavior match between rdma and sendrecv?  It won't hurt anything in the sendrecv interface, and leaves one less rock for us to trip over later.